app/index.js

```diff
@@ -7,27 +7,113 @@ export default class InfiniteScroll extends Component {
     this.state = {
       showLoader: false,
       lastScrollTop: 0,
-      actionTriggered: false
+      actionTriggered: false,
+      dragging: false,
+      pullToRefreshThresholdBreached: false
     };
+    // variables to keep track of pull down behaviour
+    this.startY = 0;
+    this.currentY = 0;
+    // will be populated in componentDidMount
+    // based on the height of the pull down element
+    this.maxPullDownDistance = 0;
+
     this.onScrollListener = this.onScrollListener.bind(this);
     this.debouncedOnScrollListener = debounce(this.onScrollListener, 150).bind(this);
+    this.onStart = this.onStart.bind(this);
+    this.onMove = this.onMove.bind(this);
+    this.onEnd = this.onEnd.bind(this);
   }
 
   componentDidMount () {
     this.el = this.props.height ? this._infScroll : window;
     this.el.addEventListener('scroll', this.debouncedOnScrollListener);
+
+    if (this.props.pullDownToRefresh) {
+      document.addEventListener('touchstart', this.onStart);
+      document.addEventListener('touchmove', this.onMove);
+      document.addEventListener('touchend', this.onEnd);
+
+      document.addEventListener('mousedown', this.onStart);
+      document.addEventListener('mousemove', this.onMove);
+      document.addEventListener('mouseup', this.onEnd);
+
+      // get BCR of pullDown element to position it above
+      this.maxPullDownDistance = this._pullDown.firstChild.getBoundingClientRect().height;
+      this.forceUpdate();
+    }
   }
 
   componentWillUnmount () {
     this.el.removeEventListener('scroll', this.debouncedOnScrollListener);
+
+    if (this.props.pullDownToRefresh) {
+      document.removeEventListener('touchstart', this.onStart);
+      document.removeEventListener('touchmove', this.onMove);
+      document.removeEventListener('touchend', this.onEnd);
+
+      document.removeEventListener('mousedown', this.onStart);
+      document.removeEventListener('mousemove', this.onMove);
+      document.removeEventListener('mouseup', this.onEnd);
+    }
   }
 
   componentWillReceiveProps (props) {
     // new data was sent in
     this.setState({
       showLoader: false,
-      actionTriggered: false
+      actionTriggered: false,
+      pullToRefreshThresholdBreached: false
+    });
+  }
+
+  onStart (evt) {
+    evt.preventDefault();
+
+    this.setState({dragging: true});
+    this.startY = evt.pageY || evt.touches[0].pageY;
```

Comment:
In render I need to show the appropriate message, on how long he needs to drag or he can release to refresh.